docs/data/base/components/modal/modal.md

```diff
@@ -0,0 +1,128 @@
+---
+product: base
+title: React Modal component
+components: ModalUnstyled
+githubLabel: 'component: modal'
+waiAria: https://www.w3.org/TR/wai-aria-practices/#dialog_modal
+packageName: '@mui/base'
+---
+
+# Modal
+
+<p class="description">The `Modal` component lets you create dialogs, popovers, lightboxes, and other elements that force the user to take action before continuing.</p>
+
+The `Modal` component renders its `children` in front of a backdrop. This lets you create an element that your users must interact with before continuing in the parent application.
+
+Features:
+
+- 🥞 Manages modal stacking when more than one is needed
+- 🔐 Creates a backdrop to disable interaction with the rest of the app
+- 🔐 Disables page scrolling while open
+- ♿️ Manages focus correctly between the modal and its parent app
+- ♿️ Adds the appropriate ARIA roles automatically
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+> **Note:** the term "modal" is sometimes used interchangeably with "dialog," but this is incorrect.
+> A dialog may be _modal_ or _nonmodal (modeless)_.
+>
+> A modal [blocks interaction with the rest of the application](https://en.wikipedia.org/wiki/Modal_window), forcing the user to take action.
+> As such, it should be used sparingly—only when the app _requires_ user input before it can continue.
+<!-- Uncomment the next line, once an unstyled dialog component is added in @mui/base -->
+<!-- If you are creating a modal dialog, the [`Dialog`](/material-ui/dialog/) component is better suited for this specific use case. -->
+
+`Modal` is a lower-level construct that is used in the following components:
+
+- [Dialog](/material-ui/react-dialog/)
+- [Drawer](/material-ui/react-drawer/)
+- [Menu](/material-ui/react-menu/)
+- [Popover](/material-ui/react-popover/)
+
+## Basic modal
+
+```js
+import ModalUnstyled from '@mui/base/ModalUnstyled';
+```
+
+{{"demo": "ModalUnstyled.js", "defaultCodeOpen": false }}
+
+## Nested modal
+
+> ⚠ **Note:** though it is possible to create nested modals—for example, a select modal within a dialog—stacking more than two at a time is discouraged.
```

Comment:
> stacking more than two at a time is discouraged.

Why? 😬

Reply:
Because they block the interaction with the elements behind them. If there are four stacked modals, and you want to go to back to the second one, you need to close two modals and make the interaction you did previously to get to the same state.

Reply:
```suggestion
> ⚠ **Note:** though it is possible to create nested modals—for example, a select modal within a dialog—stacking more than two at a time is discouraged. The main reason is because they block the interaction of elements behind them. If you stack too many, it will be needed to close all of them in order to get to the root state.
```

Nice! Tried to add the reasoning somehow to this note—feel like it's very worth it. Let me know what you and @samuelsycamore think! May need a bit of tweaking to get the words right.